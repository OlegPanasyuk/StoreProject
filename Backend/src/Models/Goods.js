module.exports = (sequalize, type) => {
    return sequalize.define( 'goods', {
        idgoods : {
            type : type.INTEGER,
            primaryKey : true,
            autoIncrement : true
        },
        name : {
            type : type.STRING(200)
        },
        description : {
            type : type.TEXT   
        }, 
        catalogue_id_catalogue : {
            type : type.INTEGER
        },
        price: {
            type: type.FLOAT
        }
    }, {
        timestamps : false,
        freezeTableName : true, 
        tableName : 'goods'
    });
};