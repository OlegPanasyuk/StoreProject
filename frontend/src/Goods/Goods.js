import React, { Component } from 'react';
import { CardColumns } from 'react-bootstrap';
import GoodsItem from '../GoodsItem/GoodsItem';
import PropTypes from 'prop-types';

class Goods extends Component {
    render() {
        let { obj } = this.props;

        let itemsGoods = Object.keys(obj).map((el, i) => {
            return (
                <GoodsItem key={`Goods-${obj[el].idgoods}-${i}`} obj={obj[el]} />
            );
        });
        if (itemsGoods.length === 0) {
            itemsGoods = 'There no any goods in categoty or select another item in menu';
        }
        return (
            <CardColumns style={{ columnGap: '1em' }}>
                {itemsGoods}
            </CardColumns>
        );
    }
}

Goods.propTypes = {
    obj: PropTypes.array
};

export default Goods;