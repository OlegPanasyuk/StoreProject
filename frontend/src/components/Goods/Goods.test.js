import React from 'react';
import { shallow, configure } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';
import PropTypes from 'prop-types';
import Goods from './Goods';

configure({ adapter: new Adapter() });

function setupComponent(props = {
    obj: [{
        idgoods: 1,
        name: 'Honda Civic',
        description: 'Lorem ipsum dolor sit amet consectetur adipisicing elit. \
        Blanditiis provident quo quaerat repellat fuga debitis nemo voluptates \
        quis voluptate, dolorem modi non? Nisi, maxime a.',
        catalogue_id_catalogue: 4
    }, {
        idgoods: 2,
        name: 'Volvo',
        description: 'Lorem ipsum dolor sit amet consectetur adipisicing elit. \
        Blanditiis provident quo quaerat repellat fuga debitis nemo voluptates \
        quis voluptate, dolorem modi non? Nisi, maxime a.',
        catalogue_id_catalogue: 4
    }]
}) {
    const component = shallow(<Goods {...props} />);
    return { component, props };
}


describe('Goods Component', () => {
    it('Component must be created and props must be correct', () => {
        const { component } = setupComponent();

        expect(Goods.propTypes.obj).toBe(PropTypes.array);
        expect(shallowToJson(component)).toMatchSnapshot();
    });
});
