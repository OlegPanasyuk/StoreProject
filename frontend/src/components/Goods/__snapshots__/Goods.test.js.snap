// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Goods Component Component must be created and props must be correct 1`] = `
<CardColumns
  style={
    Object {
      "columnGap": "1em",
    }
  }
>
  <ConnectFunction
    addItemToBacket={[Function]}
    id={1}
    key="Goods-1"
    obj={
      Object {
        "catalogue_id_catalogue": 4,
        "description": "Lorem ipsum dolor sit amet consectetur adipisicing elit.         Blanditiis provident quo quaerat repellat fuga debitis nemo voluptates         quis voluptate, dolorem modi non? Nisi, maxime a.",
        "idgoods": 1,
        "name": "Honda Civic",
      }
    }
  />
  <ConnectFunction
    addItemToBacket={[Function]}
    id={2}
    key="Goods-2"
    obj={
      Object {
        "catalogue_id_catalogue": 4,
        "description": "Lorem ipsum dolor sit amet consectetur adipisicing elit.         Blanditiis provident quo quaerat repellat fuga debitis nemo voluptates         quis voluptate, dolorem modi non? Nisi, maxime a.",
        "idgoods": 2,
        "name": "Volvo",
      }
    }
  />
</CardColumns>
`;
